packages/block-editor/src/components/block-styles/index.js

```diff
@@ -127,67 +122,35 @@ function BlockStyles( { clientId, onSwitch = noop, onHoverClassName = noop } ) {
 
 	return (
 		<div className="block-editor-block-styles">
-			{ styles.map( ( style ) => {
-				const styleClassName = replaceActiveStyle(
-					className,
-					activeStyle,
-					style
-				);
-				return (
-					<div
-						key={ style.name }
-						className={ classnames(
-							'block-editor-block-styles__item',
-							{
-								'is-active': activeStyle === style,
-							}
-						) }
-						onClick={ () => updateClassName( style ) }
-						onKeyDown={ ( event ) => {
-							if (
-								ENTER === event.keyCode ||
-								SPACE === event.keyCode
-							) {
-								event.preventDefault();
-								updateClassName( style );
-							}
-						} }
-						onMouseEnter={ () =>
-							onHoverClassName( styleClassName )
-						}
-						onMouseLeave={ () => onHoverClassName( null ) }
-						role="button"
-						tabIndex="0"
-						aria-label={ style.label || style.name }
-					>
-						<div className="block-editor-block-styles__item-preview">
-							<BlockStylePreview
-								type={ type }
-								block={ block }
-								blockName={ blockName }
-								useExample={ useExample }
-								styleClassName={ styleClassName }
-								viewportWidth={ 500 }
-							/>
-						</div>
-						<div className="block-editor-block-styles__item-label">
-							{ style.label || style.name }
-						</div>
-					</div>
-				);
-			} ) }
+			{ styles.map( ( style ) => (
+				<BlockStyleItem
+					activeStyle={ activeStyle }
+					className={ className }
+					key={ style.name }
+					onHoverClassName={ onHoverClassName }
+					style={ style }
+					type={ type }
+					updateClassName={ updateClassName }
+					{ ...rest }
+				/>
+			) ) }
 		</div>
 	);
 }
 
-function BlockStylePreview( {
-	useExample,
+function BlockStyleItem( {
 	type,
 	block,
 	blockName,
-	styleClassName,
-	viewportWidth,
+	useExample,
+	style,
+	activeStyle,
+	updateClassName,
+	className,
+	onHoverClassName,
```

Comment:
Some prop suggestions:

- activeStyle can be replaced with isActive
- `onHoverClassName` should we rename this `onHover` and accept  the style name (or object as  argument instead of the className.
- Can we rename `updateClassname` as `onSelect`, I also think the argument is not necessary since the parent  component already knows which style is being rendered?
-  It also  seems like instead of  `useExample`, `type ` and `block` we can just compute a base "block" on the p arent and pass just that block. (Unless I'm missing something, the base block  is always the same, we just alter its  className attribute per style)